File: vbnet/resistor-color/ResistorColor.vb
Public Module ResistorColor
    Dim lst() As String = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}

    Public Function ColorCode(color As String) As Integer
        For i = 0 To lst.Length - 1
            If lst(i) = color Then
                Return i
            End If
        Next
    End Function

    Public Function Colors() As String()
        Return lst
    End Function
End Module

File: swift/resistor-color/Sources/ResistorColor/ResistorColor.swift
import Foundation

enum ResistorColor: String, CaseIterable {
    case black, brown, red, orange, yellow, green, blue, violet, grey, white
    
    static func colorCode(for color: String) -> Int {
        guard let colorEnum = ResistorColor(rawValue: color.lowercased()) else {
            return -1
        }
        return ResistorColor.allCases.firstIndex(of: colorEnum)!
    }
    static var colors : [String] {
        return ResistorColor.allCases.map { $0.rawValue }
    }
}
